gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskType.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * Google Compute Engine Disk type. A disk type represents the type of disk to use, such as a
+ * {@code pd-ssd} or {@code pd-standard}.
+ */
+public final class DiskType implements Serializable {
+
+  static final Function<com.google.api.services.compute.model.DiskType, DiskType> FROM_PB_FUNCTION =
+      new Function<com.google.api.services.compute.model.DiskType, DiskType>() {
+        @Override
+        public DiskType apply(com.google.api.services.compute.model.DiskType pb) {
+          return DiskType.fromPb(pb);
+        }
+      };
+  static final Function<DiskType, com.google.api.services.compute.model.DiskType> TO_PB_FUNCTION =
+      new Function<DiskType, com.google.api.services.compute.model.DiskType>() {
+        @Override
+        public com.google.api.services.compute.model.DiskType apply(DiskType diskType) {
+          return diskType.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -944042261695072026L;
+
+  private final Long id;
+  private final DiskTypeId diskTypeId;
+  private final String creationTimestamp;
+  private final String description;
+  private final String validDiskSize;
+  private final String selfLink;
+  private final Long defaultDiskSizeGb;
+
+  static final class Builder {
+
+    private Long id;
+    private DiskTypeId diskTypeId;
+    private String creationTimestamp;
+    private String description;
+    private String validDiskSize;
+    private String selfLink;
+    private Long defaultDiskSizeGb;
+
+    private Builder() {}
+
+    Builder id(Long id) {
```

Comment:
Could this be private (since we have the static package scope "builder" method below)?